models/Document.php

```diff
@@ -779,6 +779,15 @@ public function delete()
             $service = new Document\Service;
             $service->removeTranslation($this);
 
+            // if document is registered as site remove it
+            try {
+                if (($site = Site::getByRootId($this->getId())) instanceof Site) {
```

Comment:
Sites are only possible on pages, so the right place to delete the site would be here: 
https://github.com/pimcore/pimcore/blob/cef6aa4253fa6cde359fe8644e3240aa03ec1d48/models/Document/Page.php#L83

Reply:
Ah yes, I didn't think that far... I will adapt it later